platform/server-bundle/src/main/resources/webapp/js/services.js

```diff
@@ -2,9 +2,277 @@
     'use strict';
 
     var uiServices = angular.module('uiServices', ['ngResource']);
-
     uiServices.factory('Menu', function($resource) {
         return $resource('module/menu');
     });
 
+    uiServices.service('BootstrapDialogManager', function () {
+        var modals2 = [];
+
+        this.open = function (dialog, isThisLoadingModal) {
+            modals2.push(dialog);
+            console.log("open"+modals2.length);
+            if (modals2.length > 1) {
+                modals2[modals2.length-2].close();
+            }
+            dialog.open();
+        };
+
+        this.close = function (dialog) {
+            dialog.close();
+            modals2.splice(modals2.length-1);
+            console.log("close"+modals2.length);
+            if (modals2.length > 0) {
+                modals2[modals2.length-1].open();
+            }
+        };
+
+        this.remove = function () {
+            modals2[modals2.length-1].close();
+            modals2.splice(modals2.length-1);
+            console.log("remove"+modals2.length);
+            if (modals2.length > 0) {
+                modals2[modals2.length-1].open();
+            }
+        };
+    });
+
+    uiServices.service('LoadingModal', function (BootstrapDialogManager) {
+        var open = false;
+
+        this.open = function () {
+            if (!open) {
+                var dialog = new BootstrapDialog({
+                    message: function(dialogRef){
+                        var $message = $(
+                            '<div class="splash-logo"><img src="./../../static/common/img/motech-logo.gif" alt="motech-logo"></div>' +
+                            '<div class="clearfix"></div>' +
+                            '<div class="splash-loader"><img src="./../../static/common/img/loadingbar.gif" alt="Loading..."></div>' +
+                            '<div class="clearfix"></div>' + '<br>');
+                        return $message;
+                    },
+                    closable: false,
+                    draggable: false
+                });
+                dialog.realize();
+                dialog.getModalHeader().hide();
+                dialog.getModalFooter().hide();
+                dialog.getModalContent().addClass('splash');
+                dialog.getModalContent().css('margin-top', '40%');
+                dialog.getModalBody().css('padding', '0px');
+
+                BootstrapDialogManager.open(dialog, true);
+                open = true;
+            }
+        };
+
+        this.close = function () {
+            if (open) {
+                BootstrapDialogManager.remove();
+            }
+            open = false;
+        };
+
+        this.isOpen = function () {
+            return open;
+        };
+    });
+
+    uiServices.service('Modal', function (LoadingModal, BootstrapDialogManager) {
+
+        this.alert = function () {
```

Comment:
I got really confused that this function didn't take any arguments, and then sets them further down in the code...could we move the defaultOptions out of the function to expose what you are doing with the arguments array more immediately?